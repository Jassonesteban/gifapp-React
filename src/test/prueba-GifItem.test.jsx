import { render, screen } from "@testing-library/react";
import { GifItem } from "../components/GifItem";

describe('Prueba del acrchivo gifitem <GifItem/>', () => { 

    const title = 'pokemon';
    const url = 'https://pokemon.com/pikachu';

    test('Debe hacer match con el snapshot', () => { 
        const {container} = render(<GifItem title={title} url={url}/>);
        expect(container).toMatchSnapshot();
    });

    test('debe mostrar la imagen con el url y el ALT indicado', () => { 
        render(<GifItem title={title} url={url}/>);
        //screen.debug();
        //expect(screen.getByRole('img').src).toBe(url);
        //expect(screen.getByRole('img').alt).toBe(title);

        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('Debe mostrar el titulo en el componente', () => { 
        render(<GifItem title={title} url={url}/>);
        expect(screen.getByText(title)).toBeTruthy();
    })

});