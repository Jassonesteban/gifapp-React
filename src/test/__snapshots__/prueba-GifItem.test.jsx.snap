// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del acrchivo gifitem <GifItem/> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="pokemon"
      src="https://pokemon.com/pikachu"
    />
    <p>
      pokemon
    </p>
  </div>
</div>
`;
